### ISSUE TITLE
Periodic cluster 2 - Unable to start workspace due to untracked workspace deployment 

### ISSUE THREAD

**Issue by Katka92** [redhat-developer/che-functional-tests#473]
Unable to start worksapce on cluster 2 with user ```osiotest2``` Error:
```
Error: Failed to run the workspace: "Unrecoverable event occurred: 'FailedCreate', 'Error creating: pods "workspaceagjbf0euhcjaz8e3.dockerimage-5dcc95477f-8tp75" is forbidden: exceeded quota: compute-resources, requested: limits.cpu=4,limits.memory=2147483648, used: limits.cpu=4,limits.memory=2147483648, limited: limits.cpu=6,limits.memory=3Gi', 'workspaceagjbf0euhcjaz8e3.dockerimage-5dcc95477f'"
```

I can reproduce this issue with my personal account too. 
Flaky periodic tests fails with that error too but works when creating workspace manually. Tried with Che 6 and Che 7 workspace.

**Comment by Katka92** [comment]
There is pod running but user has no workspace created. [1] I don't have permissions to manually remove this pod.

[1]
![Screenshot from 2019-04-08 07-53-27](https://user-images.githubusercontent.com/29278800/55701530-ca637d00-59d3-11e9-878a-0215a7f96380.png)

**Comment by ibuziuk** [comment]
@Katka92 could you please provide openshift events?

**Comment by Katka92** [comment]
It's pretty same as the error message in UI:
```Error creating: pods "workspacezvjqa0m7q3ggpph9.dockerimage-559cf8c67b-smjtl" is forbidden: exceeded quota: compute-resources, requested: limits.cpu=4,limits.memory=2147483648, used: limits.cpu=3906m,limits.memory=2097152k, limited: limits.cpu=6,limits.memory=3Gi
```

**Comment by Katka92** [comment]
Basically it says that it is exceeding quota limits. And IMHO this is caused by pod running in <username>-che namespace.

**Comment by ibuziuk** [comment]
well, this is a problem - quota can not be exceeded since we allow only 1 running workspace at a time. So there is a pod running in the namespace which does not correspond to any of the running workspaces and it is never terminated?

**Comment by Katka92** [comment]
Exactly. As described here https://github.com/redhat-developer/che-functional-tests/issues/473#issuecomment-480693779.

**Comment by Katka92** [comment]
I've checked the time for how long these pods are running and for both it is 3 days now (started on 5th April).

**Comment by ibuziuk** [comment]
@sleshchenko any ideas, AFIK those untracked pods should have been removed by the activity check, right?

**Comment by sleshchenko** [comment]
@ibuziuk it should if Che Server has permissions to do it.
@davidfestal You may know better credentials caching for RhChe.
@Katka92 It may be useful to see items from Che Server logs which are related to Running Pod, to search you can use workspace id that is stored deployments labels.

**Comment by Katka92** [comment]
@sleshchenko The pod is now 4 days old and I'm not able to get that old logs from rhche pod on dsaas. These logs should be in kibana, but it seems not working [1]. I don't know any other way how to get logs from that pod. If thank helps you, the id of the workspace is ``` workspace79j9wfvtjnn7s1t1 ```.

[1] https://gitlab.cee.redhat.com/dtsd/housekeeping/issues/2579

**Comment by amisevsk** [comment]
The inactivity check should be able to clean the pod without issue permissions wise, but it'll only check if the workspace is running if it's marked as `RUNNING` in the db. I'm not 100% sure how we handle failed stops, but if for some reason cleanup failed and the server marked the workspace as stopped, it would not be checked again by the activity checker.

**Comment by Katka92** [comment]
Hi guys, is there some progress? We don't have check for cluster 2 for 4 days now because of this issue.

**Comment by l0rd** [comment]
@ibuziuk @amisevsk I think we should understand the root cause of the problem but in the meantime, to unblock @Katka92, I think we should ask to clean/recreate/deprovision this namespace/user.

**Comment by ibuziuk** [comment]
@l0rd yeah, we could do this. But I will try to investiagate a bit first. Not sure I will be able to extract any relevant info though - kibana is broken on prod-preview / prod and there is no access to db. Might need to create a HK for executing some SQL queries against prod db to clarify the workspace state cc: @pbergene

**Comment by Katka92** [comment]
@ibuziuk It would be greate if you can investigate and maybe create some fix here, because there are others account affected too. 
As far as we are not using any specific tests for flaky suite, I can change the suite here to be same as in periodic jobs and we can test cluster 2 from job running for flaky tests. I'll create PR for that so we will not be stuck there and you can investigate. I'll write an info about it to daily status on ```openshiftio-che``` mattermost channel so it will be clear for everyone.

**Comment by ibuziuk** [comment]
@Katka92 could you please list all the users you are aware of which are affected by this issue?

**Comment by Katka92** [comment]
@ibuziuk ```kkanova``` and ```osiotest2```. But I think if there are two accounts broken by the same thing, there can be more accounts affected too.

**Comment by ibuziuk** [comment]
@Katka92 I have cleaned `kkanova-che` and `osiotest2-che` manually and this should unblock testing. Some differences between 2 cases:
- `kkanova-che` had only pod and deployment whereas osiotest2-che had all the resources dedicated to namespaces - deployment, pod, route, svc
- `kkanova-che` 
   - Start Time: Fri, 05 Apr 2019 15:10:22 +0000
   - Restart Count:  0
- ` osiotest2-che`
   - Running for 10d
   - Restart Count:  0

Basically, we can not proceed with any other investigation until `kubernetes.labels.app:"rhche" AND kubernetes.namespace_name:"dsaas-production"` returns no results on prod cc: @pbergene

**Comment by Katka92** [comment]
@ibuziuk I can verify - both accounts are working correctly now.

**Comment by ibuziuk** [comment]
ok, closing and tracking untracked deployment issue in rh-che - https://github.com/redhat-developer/rh-che/issues/1358
